### DISCUSSION TITLE
Use v_for to get values of nested dictionary in vuetify

### DISCUSSION THREAD

**Discussion by willy6098** [Kitware/trame#357] [Q&A] [answered]
Hello trame team!  
I am new at trame/vuetify.  
Recently trying to develop small apps with trame. To create many sliders, there is a nested dictionary which stores label, min, max and step of each slider, and I use VSlider with v_for to loop it.  
But I hit an obstacle in getting values, below is part of my Python code.  
```python
state.parameters_setting = {
    'length': {
        'Value': 0.0,
        'Min': 0,
        'Max': 10,
        'Step': 1,
        'Unit': 'cm'},
    'weight': {
        'Value': 0.1,
        'Min': 0,
        'Max': 20,
        'Step': 2,
        'Unit': 'g'}
}
vuetify.VSlider(
        v_for="(value, key) in parameters_setting",
        min=('{{parameters_setting[key]["Min"]}}', ''),
        max=('{{parameters_setting[key]["Max"]}}', ''),
        step=('{{parameters_setting[key]["Step"]}}', ''),
        label=('key',)
)
```  
With my code, the slider label works as expected, but **min, max and step** don't.  
Does anyone have experience with using VSlider v_for on nested dictionary?

**Comment by jourdain** [comment, marked as answer]
The `{{ ... }}` are meant only for text interpolation not the properties since those became plain JS rather than being vue templating logic.

So the solution will be

```python
def py_fn_update_value(slider_name, value):
    print(f"slider {slider_name} change to {value}")

vuetify.VSlider(
        v_for="(value, key) in parameters_setting",
        key="key",
        min=("value.Min",),
        max=("value.Max",),
        step=("value.Step",),
        label=("key",),
        change=(py_fn_update_value, "[key, $event]"), # assuming you want to call a function when slider change
)
```

**Reply by willy6098** [reply]
It works, thank you!

**Reply by willy6098** [reply]
I want to call a function when slider change.  
But it seems the slider value doesn't dynamically update to my state variable, which makes the called function fail.    
I had saved the key of dictionary as variable to the server state.  

```python
vuetify.VSlider(
    v_for="(value, key) in parameters_setting",
    key="key",
    v_model=("key", "value.Value"),
    change=self.execute_fun,
    min=('value.Min',),
    max=('value.Max',),
    step=('value.Step',),
    label = ('key',),
    thumb_label=True,
    thumb_size=4,
    thumb_color="#9BA6C1",
    ticks=True,
    hide_details=True,
    dense=True,
    style="max-width: 500px",
)
```

**Reply by jourdain** [reply]
The right side of the tuple on arg is the Python side and not the JS one...

`v_model=("key", "value.Value"),` translate to `server.state.key = "value.Value"`.

This is what you are looking for

```python
vuetify.VSlider(
    # ...
    v_model="value.Value",
    change="flushState('parameters_setting')",
)
```

**Reply by willy6098** [reply]
Yes, this is almost what I am looking for.
Thank you jourdain!
